go/parquet/pqarrow/file_reader.go

```diff
@@ -375,6 +375,12 @@ func (fr *FileReader) ReadRowGroups(ctx context.Context, indices, rowGroups []in
 		data.data.Release()
 	}
 
+	if ctxErr := ctx.Err(); err == nil && ctxErr != nil {
+		// if the context is in error, but we haven't set an error yet, then it means that the parent context
+		// was cancelled. In this case, we should exit early as some columns may not have been read yet.
+		err = ctxErr
+	}
```

Comment:
This should be logically equivalent, just a little simpler:
```go
if err == nil {
	err = ctx.Err()
}
```

If you're up for testing it out, [errors.Join](https://pkg.go.dev/errors#Join) would be perfect for this but I'm not 100% sure it's supported on our current version of Go.

Reply:
Thanks, done! Looks like `errors.Join` was added in 1.20 and this repo uses 1.21+, so we should be good to go